src/usr/local/www/firewall_rules_edit.php

```diff
@@ -1090,6 +1084,17 @@ function build_if_list() {
 	return($iflist);
 }
 
+// Expands missing/blank/"all" icmptypes to a list of specific types. Doesn't check validity.
+function expand_icmptype($icmptype, $ipprotocol) {
+	if (isset($icmptype) && is_string($icmptype) && strlen($icmptype) > 0 && $icmptype != "any") {
+		// $icmptype appears to contain specific icmptype(s)
+		return $icmptype;
+	} else {
+		// $icmptype has no (or blank) icmptype,or is "all"  meaning "all/any", select all subtypes
+		return implode(',', $icmptypes[$ipprotocol]['icmptypes']);
```

Comment:
If I got it right $icmptypes is a variable defined out of this function scope, then it's missing a global declaration